lambda_functions/va_profile_remove_old_opt_outs/va_profile_remove_old_opt_outs_lambda.py

```diff
@@ -3,38 +3,58 @@
 import psycopg2
 import sys
 
-# Set globals
-REMOVE_OPTED_OUT_RECORDS_QUERY = """SELECT va_profile_remove_old_opt_outs();"""
-SQLALCHEMY_DATABASE_URI = os.getenv("SQLALCHEMY_DATABASE_URI")
-logger = logging.getLogger('va_profile_remove_old_opt_outs')
+logger = logging.getLogger("va_profile_remove_old_opt_outs")
 logger.setLevel(logging.INFO)
 
-# Verify environment is setup correctly
-if SQLALCHEMY_DATABASE_URI is None:
-    logger.error("The database URI is not set.")
-    sys.exit("Couldn't connect to the database.")
-else:
-    logger.info('Execution environment prepared...')
+REMOVE_OPTED_OUT_RECORDS_QUERY = """SELECT va_profile_remove_old_opt_outs();"""
+
+# Get the database URI.  The environment variable SQLALCHEMY_DATABASE_URI is
+# set during unit testing.
+sqlalchemy_database_uri = os.getenv("SQLALCHEMY_DATABASE_URI")
+
+if sqlalchemy_database_uri is None:
+    # This should be an AWS deployment environment.  SQLALCHEMY_DATABASE_URI
+    # is not set in that case.
+
+    database_uri_path = os.getenv("DATABASE_URI_PATH")
+    if database_uri_path is None:
+        # Without this value, this code cannot know the path to the required
+        # SSM Parameter Store resource.
+        sys.exit("DATABASE_URI_PATH is not set.  Check the Lambda console.")
+
+    logger.debug("Getting the database URI from SSM Parameter Store . . .")
+    ssm_client = boto3.client("ssm")
+    ssm_response: dict = ssm_client.get_parameter(
+        Name=database_uri_path,
+        WithDecryption=True
+    )
+    logger.debug(". . . Retrieved the database URI from SSM Parameter Store.")
+    sqlalchemy_database_uri = ssm_response.get("Parameter", {}).get("Value")
+
+if sqlalchemy_database_uri is None:
+    sys.exit("Can't get the database URI.")
```

Comment:
Aside from the general idea that methods are really suppose to be "one exit", this fails to respond to VAPRO.